packages/adblocker/src/engine/bucket/cosmetic.ts

```diff
@@ -505,6 +526,10 @@ export default class CosmeticFilterBucket {
       );
     }
 
+    // Additional data for engine events
+    const matches: CosmeticFilter[] = [...rules];
```

Comment:
Do we need this? I don't see where it's used.

Reply:
It's a copy of filter matches before any of exceptions is applied. The purpose is to give a set of filters matched regardless of the exceptions.